dll/win32/shell32/brfolder.cpp

```diff
@@ -86,6 +84,49 @@ BrFolder_GetItemData(BrFolder *info, HTREEITEM hItem)
     return (BrItemData *)item.lParam;
 }
 
+static HRESULT
+BrFolder_GetChildrenEnum(
+    _In_ BrFolder *info,
+    _In_ BrItemData *pItemData,
+    _Outptr_opt_ IEnumIDList **ppEnum)
+{
+    if (!pItemData)
+        return E_FAIL;
+
+    CComPtr<IEnumIDList> pEnumIL;
+    PCUITEMID_CHILD pidlRef = pItemData->pidlChild;
+    ULONG attrs = SFGAO_HASSUBFOLDER | SFGAO_FOLDER;
+    IShellFolder *lpsf = pItemData->lpsfParent;
+    HRESULT hr = lpsf->GetAttributesOf(1, &pidlRef, &attrs);
+    if (FAILED_UNEXPECTEDLY(hr) || !(attrs & SFGAO_FOLDER))
+        return E_FAIL;
+
+    CComPtr<IShellFolder> psfChild;
+    if (_ILIsDesktop(pItemData->pidlFull))
+        hr = SHGetDesktopFolder(&psfChild);
+    else
+        hr = lpsf->BindToObject(pidlRef, NULL, IID_PPV_ARG(IShellFolder, &psfChild));
+    if (FAILED_UNEXPECTEDLY(hr))
+        return E_FAIL;
+
+    DWORD flags = BrowseFlagsToSHCONTF(info->lpBrowseInfo->ulFlags);
+    HWND hwnd = GetParent(info->hwndTreeView);
```

Comment:
I wonder whether this parent window of the treeview (which should then be the dialog itself?) could be cached in a member of `BrFolder` structure, when initializing the dialog, so that it could be simply reused here, and in the `BrFolder_Expand` function as well (where such a `GetParent()` call is also present).

Reply:
It is already cached in the struct.

Reply:
hmm... then why do we do a GetParent call here and in the other place I mention?

Reply:
Used `info->hWnd` instead.